File: frontend/console/src/api/si.js
import Axios from "@/http";

export function createSi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/si', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSi(siId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/si/${ siId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/si', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSi(siId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/si/${ siId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSiWithFlag(siId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/si/${ siId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/si/all').then(res => {
            resolve(res)
        })
    })
}

export function querySi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/si/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/si.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SI = "/config/si"
const URL_ROUTER_SI_BY_ID = "/config/si/:si-id"
const URL_ROUTER_SI_QUERY = "/config/si/query"
const URL_ROUTER_SI_ALL = "/config/si/all"

func SiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SI, config.CreateSi)
	r.GET(URL_ROUTER_SI_BY_ID, config.RetrieveSi)
	r.PUT(URL_ROUTER_SI, config.UpdateSi)
	r.DELETE(URL_ROUTER_SI_BY_ID, config.DeleteSi)
	r.PUT(URL_ROUTER_SI_BY_ID, config.DeleteSiWithFlag)
	return r
}
